src/main/java/org/springframework/data/mapping/model/KotlinCopyMethod.java

```diff
@@ -53,7 +53,7 @@
  */
 class KotlinCopyMethod {
 
-	private static final Map<Class<?>, Optional<KotlinCopyMethod>> COPY_METHOD_CACHE = new ConcurrentReferenceHashMap<>();
+	private static final Map<Class<?>, Optional<KotlinCopyMethod>> COPY_METHOD_CACHE = new ConcurrentHashMap<>();
```

Comment:
This one should remain `ConcurrentReferenceHashMap` with default soft references to not hinder GC runs.